CLONE_RESTORE.md

```diff
@@ -0,0 +1,284 @@
+     ```
+7. Delete the `harbor-operator-postgres-operator-*` pod in the `harbor-operator-ns` namespace and _wait for the pods to be recreated_(this can take some time so it might be worth deleting the postgres cluster pods also).
+   ```sh
+   kubectl -n harbor-operator-ns delete pods -l 'app.kubernetes.io/name=postgres-operator'
+   ```
+   ```sh
+   kubectl --namespace harbor-cluster delete pods -l 'application=spilo'
+   ```
+6. Go to the configured harbor url (_set as externalURL in the HarborCluster CR_) to confirm your old data has been restored.
+
+## Challenges faced while implementing this solution with Harbor Operator
+
+While this should work as it is, we encountered some bottlenecks as we implemented the solution. I'll list them and also mention ways we mitigated them:
+
+1. **Problem:** The `s3` path wasn't correct, majorly because the tools within the postgres pod (WALE, WALG, PATRONI) were outdated.
+
+   **Solution:** Update the spilo image version used by the postgres operator to a more recent version and that should fix the issue.
+
+
+2. **Problem:** `FATAL:  no pg_hba.conf entry for host "[local]", user "postgres", database "postgres", SSL off`. This was caused because the Harbor Operator hardcoded the [pg_hba.config](https://github.com/goharbor/harbor-operator/blob/master/pkg/cluster/controllers/database/generate.go#L94) and it doesn't include the `postgres` user access to make replications.
+
+   **Solution:** `This is a possible solution that needs to be pushed upstream`.
+
+    To resolve this, edit the postgres operator CR and included the following under the patroni:pg_hba section and delete the pods.
+
+    **NOTE: This is not a sustainable solution as it requires a manual intervention anytime a harbor cluster is created. A [PR](https://github.com/goharbor/harbor-operator/pull/979) has been created to resolve this issue.**
+
+    ```yaml
+    pg_hba:
+    - hostssl all             all        0.0.0.0/0 md5
+    - host    all             all        0.0.0.0/0 md5
+    - local   all             all        trust
+    - local   replication     postgres   trust
+    - hostssl replication     postgres   all       md5
+    - local   replication     standby    trust
+    - hostssl replication     standby    all       md5
+    - hostssl all             +zalandos  all       pam
+    - hostssl all             all        all       md5
+    ```
+
+3. **Problem:** The `clone_with_wale.py` script doesn't use the correct `s3` path to download the backups for restoration.
+
+   **Solution:** The `CLONE_AWS_ENDPOINT` environment variable affects the `s3` path so remove it if you have it.
+
+
+### Post Recovery Options
+
+After a database has been recovered successfully from a backup from s3, the CLONE parameters can be removed from the `pod-config` configmap or left alone, this is because the postgresql service will always pick the latest data available regardless if the CLONE parameters are specified or not.
```

Comment:
Sorry, I don't get this point. What will or won't work and how?

Reply:
I don't think this is required.